src/main/java/com/minecolonies/coremod/network/messages/server/colony/citizen/TransferItemsToCitizenRequestMessage.java

```diff
@@ -126,7 +126,7 @@ protected void onExecute(final NetworkEvent.Context ctxIn, final boolean isLogic
             int amountToRemoveFromPlayer = amountToTake - ItemStackUtils.getSize(remainingItemStack);
             while (amountToRemoveFromPlayer > 0)
             {
-                final int slot = InventoryUtils.findFirstSlotInItemHandlerWith(new InvWrapper(player.inventory), stack -> stack.isItemEqual(itemStack));
+                final int slot = InventoryUtils.findFirstSlotInItemHandlerWith(new InvWrapper(player.inventory), stack -> stack.isItemEqual(itemStack) && ItemStack.areItemStackTagsEqual(stack, itemStack));
```

Comment:
Just use ItemStackUtils equal method .

Reply:
Its like this in containers, so i just stole it from there, is that wrong?

Reply:
also there is no ItemStackUtils.equal(), or am i not seeing some third ItemStackUtils?

Reply:
done.